src/plugins/explore/public/components/visualizations/line/to_expression.ts

```diff
@@ -218,14 +222,14 @@ export const createLineBarChart = (
     },
   };
 
-  layers.push(barLayer, lineLayer);
-
   // Add threshold layer if enabled
   const thresholdLayer = createThresholdLayer(styles.thresholdLines, styles.tooltipOptions?.mode);
   if (thresholdLayer) {
-    layers.push(thresholdLayer);
+    barWithThresholdLayer.layer.push(thresholdLayer);
   }
 
+  layers.push(barWithThresholdLayer, lineLayer);
```

Comment:
Does this mean the thresholdLayer should be in front of barLayer and behind lineLayer? The original issue is that the thresholdLayer is in front of barLayer and lineLayer, which is incorrect.